apps/snitch_core/lib/core/data/schema/product.ex

```diff
@@ -0,0 +1,44 @@
+defmodule Snitch.Data.Schema.Product do
+  @moduledoc """
+  Models a Product.
+  """
+
+  use Snitch.Data.Schema
+  alias Snitch.Data.Schema.Product.NameSlug
+  alias Snitch.Data.Schema.Variant
+
+  @type t :: %__MODULE__{}
+
+  schema "snitch_products" do
+    field(:name, :string, null: false, default: "")
+    field(:description, :string)
+    field(:available_on, :utc_datetime)
+    field(:deleted_at, :utc_datetime)
+    field(:discontinue_on, :utc_datetime)
+    field(:slug, :string)
+    field(:meta_description, :string)
```

Comment:
1. I recommend wrapping all the meta fields into a `map`, making the field as `meta`.
Imagine `product.meta.description` and `product.meta.something_cool`.
2. How is `meta.description` different from `description`?
3. What is promotionable? That's a real word? How are we going to use it? AFAIK, promotions apply on product Taxons/categories.
4. What's available_on? Can that be a future date?
5. What's deleted_at? Looks like you want to implement soft deletes. But the model file is very much empty.

Things like this MUST be documented in the typedoc or moduledoc. If I'm having such doubts, everyone will.